keel-actuator/src/main/kotlin/com/netflix/spinnaker/keel/pause/ResourcePauser.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ *
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.netflix.spinnaker.keel.pause
+
+import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.api.ResourceId
+import com.netflix.spinnaker.keel.api.application
+import com.netflix.spinnaker.keel.api.id
+import com.netflix.spinnaker.keel.events.ResourceActuationPaused
+import com.netflix.spinnaker.keel.events.ResourceActuationResumed
+import com.netflix.spinnaker.keel.persistence.PausedRepository
+import com.netflix.spinnaker.keel.persistence.ResourceRepository
+import com.netflix.spinnaker.keel.persistence.ResourceStatus.PAUSED
+import org.slf4j.LoggerFactory
+import org.springframework.context.ApplicationEventPublisher
+import org.springframework.stereotype.Component
+
+@Component
+class ResourcePauser(
+  val resourceRepository: ResourceRepository,
+  val pausedRepository: PausedRepository,
+  val publisher: ApplicationEventPublisher
+) {
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+
+  fun isPaused(resource: Resource<*>): Boolean =
+    pausedRepository.applicationIsPaused(resource.application) || resourceRepository.getStatus(resource.id) == PAUSED
```

Comment:
Why doesn't the second part of this use the `pausedRepository`?

Reply:
Because we already have a pause indication with the resource status of `PAUSED`. If we want (in a future PR) we can refactor this so that all pausing is recorded in the `pausedRepository` and not done through events, but that seems outside the scope of this PR.

Reply:
What happens if I pause a resource, then update its spec? Doesn't that mean the most recent event is no longer `ResourceActuationPaused`? This seems like a kinda flaky way to represent this state.

Reply:
Ok, I'm going to handle resource pausing in the PauseRepository as well.